index.js

```diff
@@ -20,6 +22,109 @@ app.get('/info', (req, res) => {
   res.send({ serverName: packageData.name, serverVersion: packageData.version });
 });
 
+// route for displaying all products
+app.get('/products/all', (req, res) => {
+
+  var i, j;
+  var jsonOutputObject;
+
+  // Constructing JSON which lists all the products category wise
+  jsonOutputObject = '{"product_list_categorically": [';
+  for (i = 0; i < categoryData.categories.length; i++) {
```

Comment:
Generate it, and assign to a variable.
Then create a response using that. Don't combine everything into one - It will reduce your code readability.

Reply:
Done